python/tskit/text_formats.py

```diff
@@ -248,3 +250,181 @@ def build_newick(tree, *, root, precision, node_labels, include_branch_lengths):
         include_branch_lengths=include_branch_lengths,
     )
     return s + ";"
+
+
+def dump_text(
+    ts,
+    nodes,
+    edges,
+    sites,
+    mutations,
+    individuals,
+    populations,
+    provenances,
+    precision,
+    encoding,
+    base64_metadata,
+):
+    if nodes is not None:
+        print(
+            "id",
+            "is_sample",
+            "time",
+            "population",
+            "individual",
+            "metadata",
+            sep="\t",
+            file=nodes,
+        )
+        for node in ts.nodes():
+            metadata = node.metadata
+            if isinstance(metadata, bytes) and base64_metadata:
```

Comment:
Should probably make a function for this? Not point in duplicating this logic n times.
```
def get_text_metadata(metadata, base64metadata):
     if isinstance ...
         metadata = base64.b64encode(metadata).decode(encoding)
     return metadata
```